src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -139,6 +141,18 @@ public BuildData(AbstractBuild<?, ?> build, Date currentTime) {
     rootProjectDisplayName = build.getRootBuild().getDisplayName();
     rootBuildNum = build.getRootBuild().getNumber();
     buildVariables = build.getBuildVariables();
+
+    Map<String, String> env = new HashMap<String, String>();
+    List<Environment> environmentList = build.getEnvironments();
+    if (environmentList != null) {
+      for (Environment e : environmentList) {
+        if (e != null) {
+          e.buildEnvVars(env);
+          if(!env.isEmpty())
+            buildVariables.putAll(env);
+        }
+      }
+    }
```

Comment:
Can you submit a separate pull request for this change? That way this feature can be merged in regardless of the ActiveMQ support.